pkg/providers/tinkerbell/hardware/catalogue_hardware.go

```diff
@@ -141,6 +141,40 @@ func (c *Catalogue) InsertHardware(hardware *tinkv1alpha1.Hardware) error {
 	return nil
 }
 
+// RemoveHardware removes a specific hardware at a given index from the catalogue.
+func (c *Catalogue) RemoveHardware(hardware *tinkv1alpha1.Hardware, index int) error {
+	if err := c.hardwareIndex.Remove(hardware); err != nil {
+		return err
+	}
+
+	if index >= len(c.hardware) {
+		return fmt.Errorf("index out of range: %d", index)
+	}
+	c.hardware[index] = c.hardware[len(c.hardware)-1]
+	c.hardware[len(c.hardware)-1] = nil
+	c.hardware = c.hardware[:len(c.hardware)-1]
+
+	return nil
+}
+
+// RemoveHardwares removes a slice of hardwares from the catalogue.
+func (c *Catalogue) RemoveHardwares(hardware []tinkv1alpha1.Hardware) error {
+	m := make(map[string]int, len(c.hardware))
+	for i, hw := range c.hardware {
+		m[hw.Name+hw.Namespace] = i
```

Comment:
can we add a separator between name and namespace to avoid some edge cases like
```
"Name"+"Namespace"
"Nam"+"eNamespace"
```

```suggestion
		m[hw.Name+":"+hw.Namespace] = i
```

Reply:
Sure! will do